### ISSUE TITLE
Terraform try to change `enable_cross_zone_load_balancing` and `allow_overwrite` parameters

### ISSUE THREAD

**Issue by israel-tk** [terraform-providers/terraform-provider-aws#4168]
Seems an issue very similar to https://github.com/terraform-providers/terraform-provider-aws/issues/4065 but for different resources and parameters.

On plan, example:
```
  ~ module.app.aws_alb.internarl_load_balancer
      enable_cross_zone_load_balancing: "" => "false"
  ~ module.dns.aws_route53_record.internal_record
      allow_overwrite:                  "" => "true"

```
@aaithal?

**Comment by Puneeth-n** [comment]
Which version of the plugin?

**Comment by israel-tk** [comment]
AWS plugin `terraform-provider-aws_v1.13.0_x4`, terraform 0.10.8

**Comment by Puneeth-n** [comment]
Well, in your plan, it is defaulting to `false`. so you should change your configuration accordingly. Check the PR #3854 and the corresponding bug it closes #3848

**Comment by israel-tk** [comment]
But does it make sense that these attributes are empty by default? I understand they were added recently to Terraform, but the default returned by AWS should be `true` or `false` in most cases, not blank.

Isn't this similar to #4065 at all?

**Comment by Puneeth-n** [comment]
@israel-tk Oh AWS changes their API response from time to time as they wish. Sometimes they send sane defaults, sometimes `nil`. Check #4065 for example. They changed the API response twice in a day! 😅

**Comment by israel-tk** [comment]
OK so I'm closing this ticket as it's an AWS issue, not a Terraform one. :)

**Comment by israel-tk** [comment]
However according to https://docs.aws.amazon.com/elasticloadbalancing/latest/userguide/how-elastic-load-balancing-works.html "With Application Load Balancers, cross-zone load balancing is always enabled.".

I don't think the attribute `enable_cross_zone_load_balancing` should be available for application load balancers in Terraform. Is this a bug in the provider? If so, I can open another ticket about this.

**Comment by bflad** [comment]
The fix for suppressing non-applicable attributes in aws_lb diffs was made in #4032 and released in [version 1.14.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1140-april-06-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

If that fix is not working we can reopen this issue.
